lib/surface/api.ex

```diff
@@ -260,6 +260,7 @@ defmodule Surface.API do
          :ok <- validate_opts_keys(func, name, type, opts),
          :ok <- validate_opts(func, type, opts),
          :ok <- validate_required_opts(func, type, opts) do
+      maybe_warn_required_prop_with_default_value(func, type, opts, caller)
```

Comment:
how about `maybe_warn_mutually_exclusive_opts(...)` instead as we're running everything through instead of just props?